src/request/derive-address/DeriveAddress.js

```diff
@@ -22,86 +20,27 @@ class DeriveAddress {
         this._resolve = resolve;
         this._reject = reject;
 
-        /** @type {{address: Nimiq.Address, keyPath: string}?} selectedAddress */
-        this._selectedAddress = null;
-
-        /** @type {HTMLDivElement} */
-        this.$chooseIdenticonPage = (document.querySelector('#choose-identicon'));
-
         /** @type {HTMLFormElement} */
         const $passwordBox = (document.querySelector('.password-box'));
 
-        /** @type {HTMLDivElement} */
-        const $identiconSelector = (this.$chooseIdenticonPage.querySelector('.identicon-selector'));
-
-        /** @type {HTMLDivElement} */
-        this.$identicon = (this.$chooseIdenticonPage.querySelector('.identicon'));
-
-        /** @type {HTMLDivElement} */
-        this.$address = (this.$chooseIdenticonPage.querySelector('.address'));
-
-        /** @type {HTMLButtonElement} */
-        this.$confirmButton = (this.$chooseIdenticonPage.querySelector('.confirm'));
-
         // Create components
 
         this._passwordBox = new PasswordBox($passwordBox, {
             hideInput: !request.keyInfo.encrypted,
             buttonI18nTag: 'passwordbox-continue',
             hideCancel: true,
         });
-        this._identiconSelector = new DerivedIdenticonSelector($identiconSelector);
-
-        if (request.keyInfo.encrypted) {
-            // Unhide back-button
-            /** @type {HTMLElement} */
-            (this.$chooseIdenticonPage.querySelector('.page-header-back-button')).classList.remove('display-none');
-        }
 
         // Wire up logic
 
-        this._passwordBox.on(
-            PasswordBox.Events.SUBMIT,
-            async /** @param {string|undefined} password */ password => {
-                if (!(await this._onPasswordEntered(password))) return;
-                window.location.hash = DeriveAddress.Pages.CHOOSE_IDENTICON;
-            },
-        );
-
-        this._identiconSelector.on(
-            DerivedIdenticonSelector.Events.IDENTICON_SELECTED,
-            /** @param {{address: Nimiq.Address, keyPath: string}} selectedAddress */
-            selectedAddress => {
-                this._selectedAddress = selectedAddress;
-                this._openDetails(selectedAddress.address.toUserFriendlyAddress());
-            },
-        );
-
-        this._identiconSelector.on(
-            DerivedIdenticonSelector.Events.MASTER_KEY_NOT_SET,
-            this._reject.bind(this),
-        );
-
-        this.$confirmButton.addEventListener('click', () => {
-            if (!this._selectedAddress) return;
-
-            /** @type {KeyguardRequest.DeriveAddressResult} */
-            const result = {
-                address: this._selectedAddress.address.serialize(),
-                keyPath: this._selectedAddress.keyPath,
-            };
-
-            this._resolve(result);
+        this._passwordBox.on(PasswordBox.Events.SUBMIT, /** @param {string|undefined} password */ password => {
+            this._onPasswordEntered(password);
         });
```

Comment:
No need for the `{ }` code block.

Reply:
When I omit the parantheses, eslint expects a @returns type annotation, because the result of the `_onPasswordEntered` method is then returned.

Reply:
The alternative is to omit the whole JSDoc block, then ESLint is fine. But then `password` is `any`.